vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileViewModel.kt

```diff
@@ -182,4 +200,17 @@ class RoomProfileViewModel @AssistedInject constructor(
                     _viewEvents.post(RoomProfileViewEvents.ShareRoomProfile(permalink))
                 }
     }
+
+    private fun restoreEncryptionState() {
+        _viewEvents.post(RoomProfileViewEvents.Loading())
+        session.coroutineScope.launch {
+            try {
+                room.enableEncryption(force = true)
+            } catch (failure: Throwable) {
+                Timber.e(failure, "Failed to restore encryption state in room ${room.roomId}")
```

Comment:
Should post a ViewEvent to show the error in a dialog (no network, etc.)

Reply:
done